cmake/Subdirectories.cmake

```diff
@@ -12,9 +12,38 @@ include_guard()
 # Subdirectory processing
 ################################################################################
 
-# this function is building up the LY_EXTERNAL_SUBDIRS global property
+# these functions are used to build up the LY_EXTERNAL_SUBDIRS global property
+
+function(add_pal_external_subdirectories object_type object_path object_json_path)
+    set(pal_platforms ${PAL_PLATFORM_NAME})
+        list(APPEND pal_platforms ${LY_PAL_TOOLS_ENABLED})
+        foreach(platform ${pal_platforms})
+            string(TOLOWER ${platform} platform_lower)
+            o3de_restricted_path(${object_json_path} restricted_path parent_relative_path)
+            if(restricted_path)
+                o3de_pal_dir(pal_dir "${object_path}/Platform/${platform}" "${restricted_path}" "${object_path}" "${parent_relative_path}")
+                set(restricted_json_path "${pal_dir}/${object_type}_${platform_lower}.json")
+                if(EXISTS ${restricted_json_path})
+                    read_json_external_subdirs(restricted_external_subdirs ${restricted_json_path})
+                    foreach(restricted_external_subdir ${restricted_external_subdirs})
+                        file(REAL_PATH ${restricted_external_subdir} real_external_subdir BASE_DIRECTORY ${pal_dir})
+
+                        # Append external subdirectory if it is not in global property
+                        get_property(current_external_subdirs GLOBAL PROPERTY LY_EXTERNAL_SUBDIRS)
+                        if(NOT real_external_subdir IN_LIST current_external_subdirs)
+                            set_property(GLOBAL APPEND PROPERTY LY_EXTERNAL_SUBDIRS ${real_external_subdir})
+                            # Also append the project external subdirectores to the LY_EXTERNAL_SUBDIRS_ENGINE property
+                            set_property(GLOBAL APPEND PROPERTY LY_EXTERNAL_SUBDIRS_ENGINE ${real_external_subdir})
```

Comment:
We need to parameterize this property as well, so that that it sets correct manifest global property(`LY_EXTERNAL_SUBDIRS_ENGINE, LY_EXTERNAL_SUBDIRS_${project_name}, LY_EXTERNAL_SUBDIRS_O3DE_MANIFEST`)